lib/hash.h

```diff
@@ -55,7 +55,7 @@ struct hashstats
   /* number of empty hash buckets */
   _Atomic int empty;
   /* sum of squares of bucket length */
-  _Atomic uint64_t ssq;
+  _Atomic uint32_t ssq;
```

Comment:
this should be `size_t` or `atomic_uint_fast32_t`.  Some 64-bit platforms can't even atomically access a 32-bit word, meaning it'll emulate the operation with a 64-bit op and a cmpxchg loop.